pkg/remoteconfig/state/repository.go

```diff
@@ -253,6 +254,15 @@ func (r *Repository) Update(update Update) ([]string, error) {
 	return changedProducts, nil
 }
 
+// UpdateApplyStatus updates the config's metadata to reflect its processing state
+// Can be used after a call to Update() in order to tell the repository which config was acked, which
+// wasn't and which errors occurred while processing.
+func (r *Repository) UpdateApplyStatus(cfgPath string, status ApplyStatus) {
+	if m, ok := r.metadata[cfgPath]; ok {
```

Comment:
Note: be careful with races here, you don't want:
1. Update() 
2. async processing of config1
3. Update(), config1 is removed
4. UpdateApplyStatus of config1

Reply:
Agreed, since repository is not designed for concurrent use the clients using it should make sure to keep data race safety in mind. Let enrich the comment with this.

Reply:
Updated in b98455366605221dfb23e5d9d86203376b593754